block-storage/src/main/scala/coop/rchain/blockstorage/finality/LastFinalizedFileStorage.scala

```diff
@@ -61,7 +63,9 @@ object LastFinalizedFileStorage {
     for {
       lock                      <- Semaphore[F](1)
       lastFinalizedBlockHashOpt <- readLastFinalizedBlockHash(lastFinalizedPath)
-      state                     <- Cell.mvarCell[F, Option[BlockHash]](lastFinalizedBlockHashOpt)
+      state <- Cell.mvarCell[F, Map[Option[Validator], Option[BlockHash]]](
+                Map(none[Validator] -> lastFinalizedBlockHashOpt)
+              )
```

Comment:
Try to replace Cell with Ref because we want to get rid of Cell completely.

Reply:
This is removed along with DAG store